src/devices/Hx711/HX711.cs

```diff
@@ -0,0 +1,267 @@
+            PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _pinDout = pinDout;
+            _pinPdSck = pinPdSck;
+
+            _shouldDispose = shouldDispose || gpioController is null;
+            _gpioController = gpioController ?? new(pinNumberingScheme);
+
+            // Mutex for reading from the Hx711, in case multiple threads in client
+            // software try to access get values from the class at the same time.
+            _readLock = new object();
+
+            _gpioController.OpenPin(_pinPdSck, PinMode.Output);
+            _gpioController.OpenPin(_pinDout, PinMode.Input);
+
+            // Needed initialize
+            _isInitialize = false;
+
+            // Needed calibration
+            _isCalibrated = false;
+
+            _offsetFormZero = 0;
+            _tareValue = 0;
+            ReferanceUnit = 0;
+
+            _options = options ?? new Hx711Options();
+
+            _reader = new Hx711Reader(_gpioController, _options, pinDout, pinPdSck, _readLock);
+        }
+
+        /// <summary>
+        /// Load cells always return different values also based on their range and sensitivity.
+        /// For this reason, a first calibration step with a known weight is required.
+        /// You can repeat it several times to get a more precise value.
+        /// </summary>
+        /// <param name="knowWeight">Known weight currently on load cell and detected by the Hx711.</param>
+        /// <param name="numberOfReads">Number of readings to take from which to average, to get a more accurate value.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Throw if know weight have invalid value</exception>
+        public void SetCalibration(Mass knowWeight, int numberOfReads = 15)
+        {
+            if (knowWeight.Grams == 0)
+            {
+                throw new ArgumentOutOfRangeException(paramName: nameof(knowWeight), message: "Param value must be greater than zero!");
+            }
+
+            var readValue = _reader.Read(numberOfReads, _offsetFormZero);
+
+            lock (_readLock)
+            {
+                var referenceValue = readValue / knowWeight.Grams;
+
+                // If we do several calibrations, the most accurate value is the average.
+                _referenceUnitList.Add(referenceValue);
+                ReferanceUnit = _referenceUnitList.Average();
+
+                _isCalibrated = true;
+            }
+        }
+
+        /// <summary>
+        /// If you already know the reference unit between the Hx711 value and grams, you can set it and skip the calibration.
+        /// </summary>
+        /// <param name="referenceUnit">Conversion ratio between Hx711 units and grams</param>
+        public void SetReferenceUnit(double referenceUnit)
+        {
+            lock (_readLock)
+            {
+                ReferanceUnit = referenceUnit;

```

Comment:
if I read the rest of the code correctly, ReferanceUnit can't be 0 because it's used to divide a number. Maybe test that and throw here?

Reply:
Yes you are right. I add check and eventual throw of exception.